repos/system_upgrade/el7toel8/actors/checkosrelease/libraries/library.py

```diff
@@ -0,0 +1,59 @@
+import os
+
+from leapp.exceptions import StopActorExecution, StopActorExecutionError
+from leapp.libraries.common import reporting
+from leapp.libraries.stdlib import api
+from leapp.models import OSReleaseFacts
+
+
+def skip_check():
+    """ Check if has environment variable to skip this actor checks """
+    if os.getenv('LEAPP_SKIP_CHECK_OS_RELEASE'):
+        reporting.report_generic(
+            title='Skipped OS release check',
+            severity='low',
+            summary='OS release check skipped via LEAPP_SKIP_CHECK_OS_RELEASE env var.',
+        )
+        return True
+    return False
+
+
+def check_os_version(supported_version):
+    """ Check OS version and inhibit upgrade if not the same as supported ones """
+    if not isinstance(supported_version, dict):
+        api.current_logger().warning('The supported version value is invalid.')
+        raise StopActorExecution()
+
+    facts_messages = api.consume(OSReleaseFacts)
+    facts = next(facts_messages, None)
+    if list(facts_messages):
+        api.current_logger().warning('Unexpectedly received more than one OSReleaseFacts message.')
+    if not facts:
+        raise StopActorExecutionError(
+            'Could not check OS version', details={'details': 'No OSReleaseFacts facts found.'}
+        )
+
+    if facts.release_id not in supported_version:
+        reporting.report_generic(
+            title='Unsupported OS id',
```

Comment:
```suggestion
            title='Unsupported OS', 
            summary='Only RHEL is supported by the upgrade process',
```
This actor is in el7toel8 repository. This message is very generic and would be confusing to customers.